src/openrct2-ui/windows/Map.cpp

```diff
@@ -1404,18 +1410,49 @@ class MapWindow final : public Window
             widgets[WIDX_MAP].bottom = height - 1 - 14;
     }
 
+    void CalculateTextLayout()
+    {
+        utf8 buffer[512];
+        Formatter ft;
+
+        int32_t textOffset = 4 + LIST_ROW_HEIGHT;
+        _firstColumnWidth = 118;
+        for (uint32_t i = 0; i < 4; i++)
+        {
+            OpenRCT2::FormatStringLegacy(buffer, sizeof(buffer), MapLabels[i], ft.Data());
+            _firstColumnWidth = std::max(textOffset + GfxGetStringWidth(buffer, FontStyle::Medium), _firstColumnWidth);
+        }
+
+        textOffset += _firstColumnWidth + 4;
+        min_width = WW;
+        for (uint32_t i = 4; i < std::size(MapLabels); i++)
+        {
+            OpenRCT2::FormatStringLegacy(buffer, sizeof(buffer), MapLabels[i], ft.Data());
+            min_width = std::max(static_cast<int16_t>(textOffset + GfxGetStringWidth(buffer, FontStyle::Medium)), min_width);
+        }
+        width = std::max(min_width, width);
+        _recalculateScrollbars = true;
+    }
+
     uint8_t _activeTool;
```

Comment:
why is this window upside down for the member variables ???. Could you in a follow on PR make it so that member variables are at the top of the class.

Reply:
I did it in a separate commit on this PR since I figured I've been a bit slow.